src/raster/tests.rs

```diff
@@ -457,16 +458,57 @@ fn test_get_rasterband() {
 }
 
 #[test]
-fn test_get_no_data_value() {
-    let dataset = Dataset::open(fixture("tinymarble.tif")).unwrap();
-    let rasterband = dataset.rasterband(1).unwrap();
+fn test_get_no_data_value() -> Result<()> {
+    let dataset = Dataset::open(fixture("tinymarble.tif"))?;
+    let rasterband = dataset.rasterband(1)?;
     let no_data_value = rasterband.no_data_value();
     assert!(no_data_value.is_none());
 
-    // let dataset = Dataset::open(fixture!("bluemarble.tif")).unwrap();
-    // let rasterband = dataset.get_rasterband(1).unwrap();
-    // let no_data_value = rasterband.get_no_data_value();
-    // assert_eq!(no_data_value, Some(0.0));
+    let dataset = Dataset::open(fixture("labels.tif"))?;
+    let rasterband = dataset.rasterband(1)?;
+    let no_data_value = rasterband.no_data_value();
+    assert_eq!(no_data_value, Some(255.0));
+    Ok(())
+}
+
+#[test]
+#[cfg(all(major_ge_3, minor_ge_5))]
+fn test_no_data_value_i64() -> Result<()> {
+    let path = TempFixture::empty("test_no_data_value_i64.tiff");
+    {
+        let driver = DriverManager::get_driver_by_name("GTiff")?;
+        let ds = driver.create_with_band_type::<i64, _>(&path, 1, 1, 1)?;
+        let mut rasterband = ds.rasterband(1)?;
+        assert_eq!(rasterband.no_data_value_i64(), None);
+        rasterband.set_no_data_value_i64(Some(255i64))?;
+    }
+
+    let ds = Dataset::open(&path)?;
+    let rasterband = ds.rasterband(1)?;
+
+    assert_eq!(rasterband.no_data_value_i64(), Some(255i64));
```

Comment:
Should we use something larger here? I don't care that much.

Reply:
Sure, why not.

Are you ok with basically synthesizing the test data on the fly, rather than checking in files?

Reply:
Using MEM? Sure, that's a good idea. We're not testing GDAL here,.we just want to make sure we don't crash.

Reply:
It's using `TempFixture`, which does actually write to disk, but the rational is the same. Just testing bindings.